ui/admin/app/components/form/worker/create-worker-led/index.js

```diff
@@ -17,19 +17,34 @@ class Tag {
 export default class FormWorkerCreateWorkerLedComponent extends Component {
   // =services
   @service features;
+  @service browserObject;
 
   // =attributes
   @tracked generatedWorkerAuthToken;
-  @tracked clusterId;
+  @tracked clusterId = this.clusterIdFromURL;
   @tracked ipAddress;
   @tracked configFilePath;
   @tracked initialUpstreams;
   @tracked workerTags = A([]);
   @tracked newWorkerKey;
   @tracked newWorkerValue;
 
-  // =properties
+  get clusterIdFromURL() {
+    const hostname = this.browserObject.window.location.hostname;
+
+    // Match against a guid with either the prod, int, or dev hcp domain
+    const matcher =
+      /^([0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12})\.boundary\.(hashicorp\.cloud|hcp\.to|hcp\.dev)$/;
+
+    if (matcher.test(hostname)) {
+      // Grab the captured guid
+      return matcher.exec(hostname)[1];
+    }
```

Comment:
```suggestion
  const clusterIDMatcher =
    /^(?<clusterID>[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12})\.boundary\.(hashicorp\.cloud|hcp\.to|hcp\.dev)$/;

  return hostname.match(clusterIDMatcher)?.groups?.clusterID;
```

Reply:
I can use a named capturing group but I'm actually more of a fan of using the `if` since if you're not familiar with regex, it's much more obvious what's happening

Reply:
Actually I updated to without the if since it's clear once there's a comment IMO